crates/dojo-lang/src/plugin.rs

```diff
@@ -112,6 +132,77 @@ impl BuiltinDojoPlugin {
             .version_req(version_req)
             .build()
     }
+
+    fn result_with_diagnostic(
+        &self,
+        stable_ptr: SyntaxStablePtrId,
+        message: String,
+    ) -> PluginResult {
+        PluginResult {
+            code: None,
+            diagnostics: vec![PluginDiagnostic { stable_ptr, message }],
+            remove_original_item: false,
+        }
+    }
+
+    fn handle_fn(&self, db: &dyn SyntaxGroup, fn_ast: ast::FunctionWithBody) -> PluginResult {
+        let attrs = fn_ast.attributes(db).query_attr(db, "computed");
+        if attrs.is_empty() {
+            return PluginResult::default();
+        }
+        if attrs.len() != 1 {
+            return self.result_with_diagnostic(
+                attrs[0].attr(db).stable_ptr().untyped(),
+                format!("Expected one computed macro per function, got {:?}.", attrs.len()),
+            );
+        }
+        let attr = attrs[0].clone().structurize(db);
+        let args = attr.args;
+        if args.len() > 1 {
+            return self.result_with_diagnostic(
+                attr.args_stable_ptr.untyped(),
+                "Expected one arg for computed macro.\nUsage: #[computed(Position)]".into(),
+            );
+        }
+        let fn_decl = fn_ast.declaration(db);
+        let fn_name = fn_decl.name(db).text(db);
+        let params = fn_decl.signature(db).parameters(db);
+        let param_els = params.elements(db);
+        let mut model = None;
+        if args.len() == 1 {
+            let model_name = args[0].text(db);
+            model = Some(model_name.clone());
+            let model_type =
+                param_els[1].type_clause(db).ty(db).as_syntax_node().get_text(db).replace('@', "");
+            let model_type = model_type.split("::").last().unwrap();
```

Comment:
Hm I feel like there is a safer way to do this with the AST rather than operating directly on the text

Reply:
Yeah, We could probably get the last path segment.